### DISCUSSION TITLE
Unsupported Haxe version! Supported versions are 4.2.5 or newer (found 4.005) (Ubuntu 20.04)

### DISCUSSION THREAD

**Discussion by mixxxon** [HaxeFlixel/flixel#2835] [Q&A] [answered]
Greetings,

I am trying to set up the **HaxeFlixel** dev environment on **Ubuntu 20.04** (with VSC) and I followed the steps from the official guide found [here](https://haxeflixel.com/documentation/getting-started/). Everything ran smoothly until I tried to compile the "HelloWorld" example by using `lime test html5` when I got the following message: 

`/usr/share/haxe/lib/flixel/5,3,1/flixel/system/macros/FlxDefines.hx:79: characters 61-65 : Unsupported Haxe version! Supported versions are 4.2.5 or newer (found 4.005).`

Milan

**Comment by Snirozu** [comment]
[install a newer version of Haxe](https://haxe.org/download/)

**Comment by Geokureli** [comment]
as the error says, haxe version 4.0.5 is unsupported, use version 4.2.5 or higher, if you're new to flixel and haxe, there's no reason to use anything older than the latest version of haxe

**Comment by mixxxon** [comment]
Thanks, obviously I'm too tired...

**Comment by mixxxon** [comment, marked as answer]
Thank you [Geokureli](https://github.com/Geokureli) and [Snirozu](https://github.com/Snirozu) for your accurate and highly appreciated answers which can be found here: [Answer 1](https://github.com/HaxeFlixel/flixel/discussions/2835#discussioncomment-6153952) [Answer 2](https://github.com/HaxeFlixel/flixel/discussions/2835#discussioncomment-6154007).

By doing a default installation of Haxe on Ubuntu I got outdated version 4.0.5 so I performed the following steps to update it to acceptable ver. 4.2.5. As emphasized in [this document](https://haxe.org/download/linux/) it's important for the old version of Haxe/Neko to be completely removed before upgrading to a newer version. So here is how to update Haxe on Ubuntu 20.04:

Open the terminal (Ctrl+Alt+T) and type in the following:

`sudo apt-get purge --auto-remove haxe`

to completely remove Haxe and its dependencies.

Then we need to add an officially maintained repo that provides the latest versions of Haxe and Neko for all the currently supported Ubuntu versions (also usable for Ubuntu-based distributions, e.g. Linux Mint):

`sudo add-apt-repository ppa:haxe/releases -y`

Next, type in the following command

`sudo apt-get update`

to download package information from newly configured sources. And finally, perform an installation by typing in the following:

`sudo apt-get install haxe -y`

Now, there is one thing left to do and that is to update the path to the old Lib folder, so type in the terminal

`haxelib setup`

and when prompted for path, enter the path to your existing haxe lib (created by the old version) folder. Mine was:

`/usr/share/haxe/lib`

When this is set, type in

`haxe --version`

to check the version.
